amity.py

```diff
@@ -0,0 +1,285 @@
+import os
+import random
+
+from tabulate import tabulate
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from db.migration import Base, Person, Room, DatabaseCreator
+from people.people import Fellow, Staff
+from rooms.rooms import LivingSpace, Office
+
+
+class Amity(object):
+    """This class holds the system's core functionality"""
+    rooms_list = {}
+    people_list = {}
+
+    @staticmethod
+    def create_room(room_name, room_type):
+        """Create a new room and adds it to the rooms list"""
+        room_type = room_type.upper()
+        room_name = room_name.upper()
+        room_types = ["L", "l", "O", "o"]
+        if room_type not in room_types:
+            print("Dang!: Invalid room type entered. Use either O or L")
+        elif len(room_name) == 0 or len(room_type) == 0:
+            print("Make sure you enter all details. See help for more")
+        elif not room_name.isalpha():
+            print("Room name can only contain alphabets. Try again")
+        elif room_name in Amity.rooms_list.keys():
+            print("Dang! Name already taken. Enter another")
+        elif room_type == "L":
+            new_room = LivingSpace()
+            Amity.rooms_list[room_name] = {}
+            Amity.rooms_list[room_name]["Type"] = new_room.r_type
+            Amity.rooms_list[room_name]["Capacity"] = new_room.capacity
+            Amity.rooms_list[room_name]["Members"] = []
+            print("Yay! %s created successfully." % room_name)
+        elif room_type == "O":
+            new_room = Office()
+            Amity.rooms_list[room_name] = {}
+            Amity.rooms_list[room_name]["Type"] = new_room.r_type
+            Amity.rooms_list[room_name]["Capacity"] = new_room.capacity
+            Amity.rooms_list[room_name]["Members"] = []
+            print("Yay! %s created successfully." % room_name)
+        return
+
+    @staticmethod
+    def generate_random_office():
+        """Return a randomly selected office name from rooms_list"""
+        rooms = Amity.rooms_list.keys()
+        offices = []
+        for room in rooms:
+            if Amity.rooms_list[room]["Type"] == "OFFICE" and \
+            len(Amity.rooms_list[room]["Members"]) < Amity.rooms_list[room]["Capacity"]:
+                offices.append(room)
+        if len(offices) == 0:
+            return "None"
+        elif len(offices) == 1:
+            return offices[0]
+        else:
+            return random.choice(offices)
+
+    @staticmethod
+    def generate_random_living_space():
```

Comment:
this together with it's counterpart `generate_random_office` can be one..let no man put them asunder..